src/test/compile-fail/issue-18343.rs

```diff
@@ -14,6 +14,8 @@ struct Obj<F> where F: FnMut() -> u32 {
 
 fn main() {
     let o = Obj { closure: || 42 };
-    o.closure(); //~ ERROR no method named `closure` found
-    //~^ NOTE use `(o.closure)(...)` if you meant to call the function stored in the `closure` field
+    o.closure();
+    //~^ ERROR no method named `closure` found
+    //~| HELP use `(o.closure)(...)` if you meant to call the function stored in the `closure` field
+    //~| NOTE `closure` is a field storing a function, not a method
```

Comment:
can you change this to a `ui` test? I'd personally also rename it to something like `suggestion-for-field-with-fn-type-issue-18343.rs`, since `issue-18343.rs` seems like a name suitable only for "fix random ICE regression test" to me.